modules/gateway/peers_test.go

```diff
@@ -12,12 +12,25 @@ import (
 	"github.com/NebulousLabs/Sia/modules"
 )
 
+// dummyConn implements the net.Conn interface, but does not carry any actual
+// data. It is passed to muxado, because passing nil results in segfaults.
+type dummyConn struct {
```

Comment:
prefer the name mockConn or genericConn I think. Not a huge deal but we should probably settle on some naming conventions for mocking things.

mockConn -> a Conn with absolute minimum viable functionality, minimal statefullness, functions that just  return non-zero values and always err = nil.

mockSomethingConn -> a Conn that does Something. Maybe it has a stateful network, or can validate whether a signature was signed correctly by the calling function (by checking for a leading magic byte or something simple). mockVerifySigner for example.